### DISCUSSION TITLE
Y_df argument in `reconcile`

### DISCUSSION THREAD

**Discussion by ngupta23** [Nixtla/hierarchicalforecast#155] [Q&A]
I was wondering what the use of the `Y_df` argument in reconcile is. In the [following doc](https://github.com/Nixtla/hierarchicalforecast#readme), it is not used

```python
reconciled_forecasts = hrec.reconcile(Y_hat_df=forecasts_df, S=S, tags=tags)
```

In [this doc](https://nixtla.github.io/hierarchicalforecast/examples/introduction.html), it is used, but if I do not specify it, the results do not change. So it is not clear to me what the use of this argument is and when it is needed. 

```python
Y_rec_df = hrec.reconcile(Y_hat_df=Y_hat_df, Y_df=Y_fitted_df, S=S_df, tags=tags)
```

Any further details would be helpful. Thanks!

**Comment by ngupta23** [comment]
So, it looks like `Y_df` is only needed for certain reconciliation methods. Is this understanding correct?
If so, should this contain the actual Y values or the in-sample predictions?

**Reply by kdgutier** [reply]
Hi @ngupta23,

As you mentioned some mean/probabilistic reconciliation methods require `Y_df` insample predictions`y_hat` and target `y`.
Here is the reconciliation methods catalog: https://nixtla.github.io/hierarchicalforecast/methods.html

Here is a usage example of such methods, where we use `StatsForecast`'s fitted values:
https://github.com/Nixtla/hierarchicalforecast/blob/main/nbs/examples/TourismLarge-Evaluation.ipynb

**Reply by ngupta23** [reply]
Going through this, but have some questions.

For Middle out, why are the in-sample predictions required for `forecast_proportions` (they are not needed for top-down)?

https://nixtla.github.io/hierarchicalforecast/methods.html#middleout.fit_predict

![image](https://user-images.githubusercontent.com/33585645/211061494-6d5ffcc9-516d-4113-be3b-a251cc0698f6.png)

![image](https://user-images.githubusercontent.com/33585645/211061888-85ff485a-6b39-4ee4-aa93-94fec7c3fb06.png)

**Reply by ngupta23** [reply]
Maybe it is a typo - "Only" should be changed to "Optional" in the Middle-Out section

**Reply by kdgutier** [reply]
Thanks for the comment, we will update the documentation of the `TopDown` method, to mention it.
The documentation needs an overall revision. It is currently in its 0.0 version
:)

Please let us know if you find other things.
One thing we definitely need to improve is the introduction to the hierarchical forecast structures and the `S_df` inputs.

**Reply by ngupta23** [reply]
> we will update the documentation of the `TopDown` method, to mention it

So just to confirm, is the typo in "Top Down" or "Middle Out"? I would think "Middle Out" since you should not need in-sample when using forecast proportions.
